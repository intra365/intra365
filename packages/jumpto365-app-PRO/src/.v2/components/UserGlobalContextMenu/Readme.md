# Not logged in

```js
<UserGlobalContextMenu about="🍕" />
```

# Logged in

```js
<UserGlobalContextMenu isAuthenticated userName="Niels Gregers Johansen - jumpto365.com" userId="niels@hexatown.com" />
```