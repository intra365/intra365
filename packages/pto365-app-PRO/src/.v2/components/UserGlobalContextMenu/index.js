import React, { Component } from "react";

import Translate from "../../../../src/components/Sidebar/translation";

import PropTypes from "prop-types";

import { ContextualMenuItemType } from "office-ui-fabric-react/lib/ContextualMenu";
import {
  PrimaryButton,
  DefaultButton,
  IconButton
} from "office-ui-fabric-react/lib/Button";
import { TextField, Toggle } from "office-ui-fabric-react";
import Languages from "../../../components/Sidebar/Languages";
import packageData from "../../../../package.json";
import Jumpto365App, { EditorActions } from "../_Contexts/Jumpto365App";
import { Router, Match, Location, navigate } from "@reach/router";
import {
  Dialog,
  DialogType,
  DialogFooter
} from "office-ui-fabric-react/lib/Dialog";
import { EditorExperienceContext } from "../../logic/EditorExperience/EditorExperienceContext";
/**
 * Describe overall purpose of the component
 *
 * @export
 * @class UserGlobalContextMenu
 * @extends {Component}
 */
export default class UserGlobalContextMenu extends Component {
  state = {};
  static propTypes = {
    userName: PropTypes.string, // sample - remove or rename
    userId: PropTypes.string,
    doLogout: PropTypes.func,
    doLogin: PropTypes.func,
    isAuthenticated: PropTypes.bool,
    isMobile: PropTypes.string
  };

  constructor(props) {
    super(props);
  }

  /**
   * Required method return the output of the component
   *
   * @returns
   * @memberof UserGlobalContextMenu
   */
  render() {
    var context = this.props.globalContext;
    var userSettings =
      context && context.me && context.me.JSON
        ? JSON.parse(context.me.JSON)
        : {};
    var canUpload = userSettings.canUpload;

    var canEdit = userSettings.canEdit && context.editor;
    var editMode = context.editMode;

    var isDeveloper = userSettings.developer;
    var canShare = false;

    var hasContextMenuAction = context.hostSettings
      ? context.hostSettings.hasContextMenuAction
      : true;

    if (!hasContextMenuAction) {
      return (
        <DefaultButton
          data-automation-id="jumptoapp"
          text={
            context.hostSettings && context.hostSettings.appText
              ? context.hostSettings.appText
              : "jumpto365 App"
          }
          iconProps={{ iconName: "NavigateExternalInline" }}
          onClick={() => {
            navigate(
              context.hostSettings && context.hostSettings.appURL
                ? context.hostSettings.appURL
                : "https://app.jumpto365.com"
            );
          }}
          style={{ backgroundColor: "#ffffff" }}
          splitButtonMenuButton={{ backgroundColor: "#ffffff" }}
          splitButtonContainer={{ backgroundColor: "#ffffff" }}
        />
      );
    }
    function prompt(id) {
      return Jumpto365App.prompt(id, context);
    }

    function doEmit(func) {
      if (!func != null) return;
      func();
    }
    function languageLink(text, language) {
      return {
        key: language,
        name: text,
        onClick: () => {
          Jumpto365App.setLanguage(context, language);

          navigate(`/periodictable/${language}`);
        }
      };
    }
    function internalLink(text, path) {
      return {
        key: language,
        name: text,
        onClick: () => {
          navigate(`${path}`);
        }
      };
    }

    var loggedInMenu = [];
    loggedInMenu.push({
      key: "me",
      name: `Me`,
      icon: "PlayerSettings",
      iconProps: { iconName: "AccountManagement" },
      onClick: () => {
        navigate("/me");
      }
    });
    if (isDeveloper) {
      loggedInMenu.push({
        key: "dev",
        name: `Developer`,
        icon: "PlayerSettings",
        iconProps: { iconName: "PlayerSettings" },
        onClick: () => {
          navigate("/dev");
        }
      });
    }
    if (canEdit) {
      loggedInMenu.push({
        key: "dev",
        name: `Accounts`,
        iconProps: { iconName: "Accounts" },
        onClick: () => {
          navigate("/buy");
        }
      });
    }
    // {
    //     key:"profile",
    //     name: prompt("37"),
    //     icon:"Info",
    //     onClick: () => {
    //         navigate("/beta/me")
    //         }
    // },

    loggedInMenu.push({
      key: "content",
      name: "Content",
      disabled: false,
      hidden: !canUpload,
      iconProps: { iconName: "DocumentManagement" },
      subMenuProps: {
        items: [
          // {
          //     key:"newtable",
          //     name: `New table`,

          //     iconProps: { iconName: 'PageAdd' },
          //     onClick: () => {
          //         navigate("/new/table")
          //         }

          //     } ,

          {
            key: "articles",
            name: `Articles`,

            iconProps: { iconName: "Articles" },
            onClick: () => {
              navigate("/-/articles");
            }
          },
          // {
          //     key:"icons",
          //     name: `Icons`,

          //     iconProps: { iconName: 'PhotoCollection' },
          //     subMenuProps: {
          //         items:   [
          //             {

          //                 key:"gallery",
          //                 text:"Gallery",
          //                 iconProps:{ iconName: "ImageSearch"},

          //                 onClick:() => {
          //                     navigate("/icons")
          //                     }

          //                 }   ,
          //             {

          //                 key:"rootupload",
          //                 text:"Upload",
          //                 iconProps:{ iconName: "Upload"},

          //                 onClick:() => {
          //                     navigate("/icons/upload")
          //                     }

          //                 }
          //         ] },
          //     onClick: () => {
          //         navigate("/icons")
          //         }

          //     }
          //     ,
          {
            key: "Tables",
            name: `My Tables`,

            iconProps: { iconName: "Database" },
            onClick: () => {
              navigate("/database/mytables");
            }
          }
          // ,{

          // key:"rootupload",
          // text:"Upload",
          // iconProps:{ iconName: "Upload"},

          // onClick:() => {
          //     navigate("/upload")
          //     }

          // }
        ]
      }
    });

    loggedInMenu.push({
      key: "Logout",
      name: `${prompt("36")} (${this.props.userId})`,
      icon: "Clear",
      iconProps: { iconName: "UserRemove" },
      onClick: () => {
        Jumpto365App.logout();
      }
    });
    var userMenu = !this.props.isAuthenticated ? (
      <DefaultButton
        data-automation-id="test"
        text={prompt("25")}
        onClick={() => {
          Jumpto365App.login();
        }}
        style={{ backgroundColor: "#ffffff" }}
        splitButtonMenuButton={{ backgroundColor: "#ffffff" }}
        splitButtonContainer={{ backgroundColor: "#ffffff" }}
      />
    ) : (
      <IconButton
        data-automation-id="test"
        text={prompt("59") /* "Options" */}
        iconProps={{ iconName: "Settings" }}
        split={true}
        splitButtonAriaLabel={
          "Like more information before you decide to logout?"
        }
        aria-roledescription={"split button"}
        style={{
          backgroundColor: "#ffffff",
          paddingLeft: "0px",
          paddingRight: "0px"
        }}
        menuProps={{ items: loggedInMenu }}
      />
    );
    var buttonEdit = () => {
      return (
        <IconButton
          data-automation-id="rootedit"
          text=""
          iconProps={{ iconName: context.editMode ? "EditSolid12" : "Edit" }}
          split={false}
          splitButtonAriaLabel={""}
          aria-roledescription={"split button"}
          style={{
            backgroundColor: "#ffffff",
            paddingLeft: "0px",
            paddingRight: "0px"
          }}
          onClick={() => {
            if (!context.editMode && context.me && context.me.domain) {
              // navigate(`/@/${context.me.upn}/-`)
            }
            Jumpto365App.toogleEdit(context);
          }}
        />
      );
    };

    var buttonPublish = () => {
      return (
        <IconButton
          hidden={!context.editMode}
          Xdisabled={true}
          data-automation-id="explore"
          style={{
            backgroundColor: "#ffffff",
            paddingLeft: "0px",
            paddingRight: "0px"
          }}
          iconProps={{ iconName: "WebPublish" }}
          text="Publish"
          onClick={() => {
            context.editor.action(EditorActions.Publish);
          }}
          split={context.editMode}
          style={{ paddingLeft: "0px", paddingRight: "0px" }}
          splitButtonMenuButton={{ backgroundColor: "#ffffff" }}
          splitButtonContainer={{ backgroundColor: "#ffffff" }}
          menuProps={{
            items: [
              {
                key: "publishGlobe",
                text: "Jumpto365",
                iconProps: { iconName: "Globe" },
                onClick: () => {
                  context.editor.action(EditorActions.Publish);
                }
              },
              {
                key: "publishSharePoint",
                text: "SharePoint ðŸ’° ðŸ”’ ",
                disabled: true,
                iconProps: { iconName: "SharepointLogo" },
                onClick: () => {
                  context.editor.action(EditorActions.PublishSharePoint);
                }
              }
            ]
          }}
        />
      );
    };
    var buttonUpload = () => {
      return (
        <IconButton
          data-automation-id="rootupload"
          text=""
          disabled={context.editMode}
          iconProps={{ iconName: "Upload" }}
          split={false}
          splitButtonAriaLabel={""}
          aria-roledescription={"split button"}
          style={{
            backgroundColor: "#ffffff",
            paddingLeft: "0px",
            paddingRight: "0px"
          }}
          onClick={() => {
            navigate("/upload");
          }}
        />
      );
    };

    var buttonSave = () => {
      return (
        <IconButton
          hidden={!context.editMode}
          data-automation-id="save"
          text="Save"
          iconProps={{ iconName: "Save" }}
          split={context.editMode}
          splitButtonAriaLabel={""}
          aria-roledescription={"split button"}
          style={{
            backgroundColor: "#ffffff",
            paddingLeft: "0px",
            paddingRight: "0px"
          }}
          splitButtonMenuButton={{ backgroundColor: "#ffffff" }}
          splitButtonContainer={{ backgroundColor: "#ffffff" }}
          menuProps={{
            items: [
              {
                disable: true,
                key: "save2",
                text: "Save",
                iconProps: { iconName: "Save" },
                onClick: () => {
                  context.editor.action(EditorActions.Save);
                }
              },
              {
                disable: true,
                key: "emailMessage",
                text: "Save a copy",
                iconProps: { iconName: "SaveAs" },
                onClick: () => {
                  this.setState({ showSaveAsDialogue: true });
                }
              }
            ]
          }}
          onClick={() => {
            context.editor.action(EditorActions.Save);
          }}
        />
      );
    };

    var dialogSaveAs = () => {
      return (
        <Dialog
          hidden={!this.state.showSaveAsDialogue}
          onDismiss={() => {
            this.setState({ showSaveAsDialogue: false });
          }}
          dialogContentProps={{
            type: DialogType.normal,
            title: "Save a copy",
            xsubText:
              "Your Inbox has changed. No longer does it include favorites, it is a singular destination for your emails."
          }}
          modalProps={{
            isBlocking: true,
            containerClassName: "ms-dialogMainOverride"
          }}
        >
          <TextField
            placeholder="Name"
            ariaLabel="Icon"
            onChange={(e, v) => {
              context.editor.validateSaveAs(v).then(result => {
                if (result.isValid) {
                  this.setState({ name: v });
                } else {
                  this.setState({ conflict: result.message });
                }
              });
            }}
          />
          {this.state.conflict}
          <DialogFooter>
            <PrimaryButton
              disabled={!this.state.name}
              onClick={() => {
                context.editor.action(EditorActions.SaveAs, this.state.name);
              }}
              text="Save"
            />
            <DefaultButton
              onClick={() => {
                this.setState({ showSaveAsDialogue: false });
              }}
              text="Cancel"
            />
          </DialogFooter>
        </Dialog>
      );
    };
    return (
      <div style={{ marginTop: "4px" }}>
        {dialogSaveAs()}

        {canEdit && editMode && buttonSave()}
        {canEdit && editMode && buttonPublish()}
        {/* {canEdit && buttonEdit()} */}
        {userSettings.canEdit &&
        <EditorExperienceContext.Consumer>
          {editor => (

           
            <IconButton
              data-automation-id="rootedit"
              text=""
              iconProps={{ iconName: editor.on ? "EditSolid12" : "Edit" }}
              split={false}
              splitButtonAriaLabel={""}
              aria-roledescription={"split button"}
              style={{
                backgroundColor: "#ffffff",
                paddingLeft: "0px",
                paddingRight: "0px"
              }}
              onClick={() => {
                editor.toggle();
              }}
            />
          )}
        </EditorExperienceContext.Consumer>}
        {/* {canUpload && buttonUpload()} */}
        {userMenu}
      </div>
    );
  }
}
