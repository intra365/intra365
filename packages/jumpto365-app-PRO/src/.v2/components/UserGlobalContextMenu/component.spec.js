import React, { Component } from 'react';

import UserGlobalContextMenu from '.'

test('Check that it renders', () => {

    const wrapper = shallow(<UserGlobalContextMenu about="🍕"/>);

    expect(wrapper).toMatchSnapshot();
 
}

)