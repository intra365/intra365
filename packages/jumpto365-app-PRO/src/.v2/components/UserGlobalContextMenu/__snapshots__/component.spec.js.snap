// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check that it renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserGlobalContextMenu
    about="üçï"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <CustomizedDefaultButton
          aria-roledescription="split button"
          data-automation-id="share"
          iconProps={
                    Object {
                              "iconName": "Share",
                            }
          }
          menuProps={
                    Object {
                              "items": Array [
                                Object {
                                  "disabled": false,
                                  "icon": "Embed",
                                  "key": "embed",
                                  "name": "[[NOCONTEXT:14]]",
                                  "onClick": [Function],
                                },
                              ],
                            }
          }
          split={true}
          splitButtonAriaLabel=""
          style={
                    Object {
                              "backgroundColor": "#ffffff",
                              "paddingLeft": "0px",
                              "paddingRight": "0px",
                            }
          }
          text="[[NOCONTEXT:35]]"
/>,
        <CustomizedDefaultButton
          aria-roledescription="split button"
          data-automation-id="language"
          iconProps={
                    Object {
                              "iconName": "LocaleLanguage",
                            }
          }
          menuProps={
                    Object {
                              "items": Array [
                                Object {
                                  "key": "cs",
                                  "name": "ƒåe≈°tina",
                                  "onClick": [Function],
                                },
                                Object {
                                  "key": "da",
                                  "name": "Dansk",
                                  "onClick": [Function],
                                },
                                Object {
                                  "key": "de",
                                  "name": "Deutsche",
                                  "onClick": [Function],
                                },
                                Object {
                                  "key": "en",
                                  "name": "English",
                                  "onClick": [Function],
                                },
                                Object {
                                  "key": "es",
                                  "name": "Espa√±ol",
                                  "onClick": [Function],
                                },
                                Object {
                                  "key": "fi",
                                  "name": "Suomi",
                                  "onClick": [Function],
                                },
                                Object {
                                  "key": "fr",
                                  "name": "Fran√ßaise",
                                  "onClick": [Function],
                                },
                                Object {
                                  "key": "he",
                                  "name": "◊¢÷¥◊ë◊®÷¥◊ô◊™",
                                  "onClick": [Function],
                                },
                                Object {
                                  "key": "hu",
                                  "name": "Magyar",
                                  "onClick": [Function],
                                },
                                Object {
                                  "key": "is",
                                  "name": "√çslensk",
                                  "onClick": [Function],
                                },
                                Object {
                                  "key": "ja",
                                  "name": "Êó•Êú¨Ë™û",
                                  "onClick": [Function],
                                },
                                Object {
                                  "key": "nl",
                                  "name": "Nederlandse",
                                  "onClick": [Function],
                                },
                                Object {
                                  "key": "no",
                                  "name": "Norsk",
                                  "onClick": [Function],
                                },
                                Object {
                                  "key": "pt",
                                  "name": "Portugu√™s",
                                  "onClick": [Function],
                                },
                                Object {
                                  "key": "sv",
                                  "name": "Svenska",
                                  "onClick": [Function],
                                },
                              ],
                            }
          }
          split={true}
          splitButtonAriaLabel=""
          style={
                    Object {
                              "backgroundColor": "#ffffff",
                              "paddingLeft": "0px",
                              "paddingRight": "0px",
                            }
          }
          text="[[NOCONTEXT:5]]"
/>,
        <CustomizedDefaultButton
          data-automation-id="explore"
          iconProps={
                    Object {
                              "iconName": "ExploreData",
                            }
          }
          onClick={[Function]}
          primary={true}
          style={
                    Object {
                              "backgroundColor": "orange",
                              "paddingLeft": "0px",
                              "paddingRight": "0px",
                            }
          }
          text="[[NOCONTEXT:34]] -  Preview 11"
/>,
        null,
      ],
      "style": Object {
        "marginTop": "4px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "aria-roledescription": "split button",
          "data-automation-id": "share",
          "iconProps": Object {
            "iconName": "Share",
          },
          "menuProps": Object {
            "items": Array [
              Object {
                "disabled": false,
                "icon": "Embed",
                "key": "embed",
                "name": "[[NOCONTEXT:14]]",
                "onClick": [Function],
              },
            ],
          },
          "split": true,
          "splitButtonAriaLabel": "",
          "style": Object {
            "backgroundColor": "#ffffff",
            "paddingLeft": "0px",
            "paddingRight": "0px",
          },
          "text": "[[NOCONTEXT:35]]",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "aria-roledescription": "split button",
          "data-automation-id": "language",
          "iconProps": Object {
            "iconName": "LocaleLanguage",
          },
          "menuProps": Object {
            "items": Array [
              Object {
                "key": "cs",
                "name": "ƒåe≈°tina",
                "onClick": [Function],
              },
              Object {
                "key": "da",
                "name": "Dansk",
                "onClick": [Function],
              },
              Object {
                "key": "de",
                "name": "Deutsche",
                "onClick": [Function],
              },
              Object {
                "key": "en",
                "name": "English",
                "onClick": [Function],
              },
              Object {
                "key": "es",
                "name": "Espa√±ol",
                "onClick": [Function],
              },
              Object {
                "key": "fi",
                "name": "Suomi",
                "onClick": [Function],
              },
              Object {
                "key": "fr",
                "name": "Fran√ßaise",
                "onClick": [Function],
              },
              Object {
                "key": "he",
                "name": "◊¢÷¥◊ë◊®÷¥◊ô◊™",
                "onClick": [Function],
              },
              Object {
                "key": "hu",
                "name": "Magyar",
                "onClick": [Function],
              },
              Object {
                "key": "is",
                "name": "√çslensk",
                "onClick": [Function],
              },
              Object {
                "key": "ja",
                "name": "Êó•Êú¨Ë™û",
                "onClick": [Function],
              },
              Object {
                "key": "nl",
                "name": "Nederlandse",
                "onClick": [Function],
              },
              Object {
                "key": "no",
                "name": "Norsk",
                "onClick": [Function],
              },
              Object {
                "key": "pt",
                "name": "Portugu√™s",
                "onClick": [Function],
              },
              Object {
                "key": "sv",
                "name": "Svenska",
                "onClick": [Function],
              },
            ],
          },
          "split": true,
          "splitButtonAriaLabel": "",
          "style": Object {
            "backgroundColor": "#ffffff",
            "paddingLeft": "0px",
            "paddingRight": "0px",
          },
          "text": "[[NOCONTEXT:5]]",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data-automation-id": "explore",
          "iconProps": Object {
            "iconName": "ExploreData",
          },
          "onClick": [Function],
          "primary": true,
          "style": Object {
            "backgroundColor": "orange",
            "paddingLeft": "0px",
            "paddingRight": "0px",
          },
          "text": "[[NOCONTEXT:34]] -  Preview 11",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <CustomizedDefaultButton
            aria-roledescription="split button"
            data-automation-id="share"
            iconProps={
                        Object {
                                    "iconName": "Share",
                                  }
            }
            menuProps={
                        Object {
                                    "items": Array [
                                      Object {
                                        "disabled": false,
                                        "icon": "Embed",
                                        "key": "embed",
                                        "name": "[[NOCONTEXT:14]]",
                                        "onClick": [Function],
                                      },
                                    ],
                                  }
            }
            split={true}
            splitButtonAriaLabel=""
            style={
                        Object {
                                    "backgroundColor": "#ffffff",
                                    "paddingLeft": "0px",
                                    "paddingRight": "0px",
                                  }
            }
            text="[[NOCONTEXT:35]]"
/>,
          <CustomizedDefaultButton
            aria-roledescription="split button"
            data-automation-id="language"
            iconProps={
                        Object {
                                    "iconName": "LocaleLanguage",
                                  }
            }
            menuProps={
                        Object {
                                    "items": Array [
                                      Object {
                                        "key": "cs",
                                        "name": "ƒåe≈°tina",
                                        "onClick": [Function],
                                      },
                                      Object {
                                        "key": "da",
                                        "name": "Dansk",
                                        "onClick": [Function],
                                      },
                                      Object {
                                        "key": "de",
                                        "name": "Deutsche",
                                        "onClick": [Function],
                                      },
                                      Object {
                                        "key": "en",
                                        "name": "English",
                                        "onClick": [Function],
                                      },
                                      Object {
                                        "key": "es",
                                        "name": "Espa√±ol",
                                        "onClick": [Function],
                                      },
                                      Object {
                                        "key": "fi",
                                        "name": "Suomi",
                                        "onClick": [Function],
                                      },
                                      Object {
                                        "key": "fr",
                                        "name": "Fran√ßaise",
                                        "onClick": [Function],
                                      },
                                      Object {
                                        "key": "he",
                                        "name": "◊¢÷¥◊ë◊®÷¥◊ô◊™",
                                        "onClick": [Function],
                                      },
                                      Object {
                                        "key": "hu",
                                        "name": "Magyar",
                                        "onClick": [Function],
                                      },
                                      Object {
                                        "key": "is",
                                        "name": "√çslensk",
                                        "onClick": [Function],
                                      },
                                      Object {
                                        "key": "ja",
                                        "name": "Êó•Êú¨Ë™û",
                                        "onClick": [Function],
                                      },
                                      Object {
                                        "key": "nl",
                                        "name": "Nederlandse",
                                        "onClick": [Function],
                                      },
                                      Object {
                                        "key": "no",
                                        "name": "Norsk",
                                        "onClick": [Function],
                                      },
                                      Object {
                                        "key": "pt",
                                        "name": "Portugu√™s",
                                        "onClick": [Function],
                                      },
                                      Object {
                                        "key": "sv",
                                        "name": "Svenska",
                                        "onClick": [Function],
                                      },
                                    ],
                                  }
            }
            split={true}
            splitButtonAriaLabel=""
            style={
                        Object {
                                    "backgroundColor": "#ffffff",
                                    "paddingLeft": "0px",
                                    "paddingRight": "0px",
                                  }
            }
            text="[[NOCONTEXT:5]]"
/>,
          <CustomizedDefaultButton
            data-automation-id="explore"
            iconProps={
                        Object {
                                    "iconName": "ExploreData",
                                  }
            }
            onClick={[Function]}
            primary={true}
            style={
                        Object {
                                    "backgroundColor": "orange",
                                    "paddingLeft": "0px",
                                    "paddingRight": "0px",
                                  }
            }
            text="[[NOCONTEXT:34]] -  Preview 11"
/>,
          null,
        ],
        "style": Object {
          "marginTop": "4px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "aria-roledescription": "split button",
            "data-automation-id": "share",
            "iconProps": Object {
              "iconName": "Share",
            },
            "menuProps": Object {
              "items": Array [
                Object {
                  "disabled": false,
                  "icon": "Embed",
                  "key": "embed",
                  "name": "[[NOCONTEXT:14]]",
                  "onClick": [Function],
                },
              ],
            },
            "split": true,
            "splitButtonAriaLabel": "",
            "style": Object {
              "backgroundColor": "#ffffff",
              "paddingLeft": "0px",
              "paddingRight": "0px",
            },
            "text": "[[NOCONTEXT:35]]",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "aria-roledescription": "split button",
            "data-automation-id": "language",
            "iconProps": Object {
              "iconName": "LocaleLanguage",
            },
            "menuProps": Object {
              "items": Array [
                Object {
                  "key": "cs",
                  "name": "ƒåe≈°tina",
                  "onClick": [Function],
                },
                Object {
                  "key": "da",
                  "name": "Dansk",
                  "onClick": [Function],
                },
                Object {
                  "key": "de",
                  "name": "Deutsche",
                  "onClick": [Function],
                },
                Object {
                  "key": "en",
                  "name": "English",
                  "onClick": [Function],
                },
                Object {
                  "key": "es",
                  "name": "Espa√±ol",
                  "onClick": [Function],
                },
                Object {
                  "key": "fi",
                  "name": "Suomi",
                  "onClick": [Function],
                },
                Object {
                  "key": "fr",
                  "name": "Fran√ßaise",
                  "onClick": [Function],
                },
                Object {
                  "key": "he",
                  "name": "◊¢÷¥◊ë◊®÷¥◊ô◊™",
                  "onClick": [Function],
                },
                Object {
                  "key": "hu",
                  "name": "Magyar",
                  "onClick": [Function],
                },
                Object {
                  "key": "is",
                  "name": "√çslensk",
                  "onClick": [Function],
                },
                Object {
                  "key": "ja",
                  "name": "Êó•Êú¨Ë™û",
                  "onClick": [Function],
                },
                Object {
                  "key": "nl",
                  "name": "Nederlandse",
                  "onClick": [Function],
                },
                Object {
                  "key": "no",
                  "name": "Norsk",
                  "onClick": [Function],
                },
                Object {
                  "key": "pt",
                  "name": "Portugu√™s",
                  "onClick": [Function],
                },
                Object {
                  "key": "sv",
                  "name": "Svenska",
                  "onClick": [Function],
                },
              ],
            },
            "split": true,
            "splitButtonAriaLabel": "",
            "style": Object {
              "backgroundColor": "#ffffff",
              "paddingLeft": "0px",
              "paddingRight": "0px",
            },
            "text": "[[NOCONTEXT:5]]",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data-automation-id": "explore",
            "iconProps": Object {
              "iconName": "ExploreData",
            },
            "onClick": [Function],
            "primary": true,
            "style": Object {
              "backgroundColor": "orange",
              "paddingLeft": "0px",
              "paddingRight": "0px",
            },
            "text": "[[NOCONTEXT:34]] -  Preview 11",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
